### ISSUE TITLE
Strange behaviour with Pytest when running script is imported as module in another test file.

### ISSUE THREAD

**Issue by weiss-d** [kvas-it/pytest-console-scripts#46]
Hi, everyone!
Let's assume I have a project with the following structue:
```
.
├── mock_check
│   ├── __init__.py
│   └── my_script.py
└── test_my_script.py
```
Files contents are:
```python
# mock_check/my_script.py
import requests
import sys

def some_func():
    return 1

print(sys.argv)
print(requests.get("http://fake.url").json())

```

```python
# test_my_script.py
import pytest


@pytest.fixture
def mock_requests_get(mocker):
    mock = mocker.patch("requests.get")
    mock.return_value.json.return_value = "test_output"
    return mock


@pytest.mark.script_launch_mode('inprocess')
def test_mock_check(mock_requests_get, script_runner):
    ret = script_runner.run("mock_check/my_script.py", print_result=True, shell=True)
    assert ret.success
    assert ret.stdout == "['mock_check/my_script.py']\ntest_output\n"
    assert ret.stderr == ''
```

When I run `pytest` this single test passes.

Suppose I want to test `some_func()` separately.
So I make a new file:
```python
# test_import_my_script.py
from mock_check.my_script import some_func

def test_some_func():
    assert some_func() == 1
```
And project structuer becames this:
```
.
├── mock_check
│   ├── __init__.py
│   └── my_script.py
├── test_import_my_script.py
└── test_mock_check.py
```

Now when I run `pytest` it gathers both files and command terminates with error:
```
ERROR test_import_my_script.py - requests.exceptions.ConnectionError: HTTPConnectionPool(host='fake.url', port=80): Max retries exceeded...
```
which means that mock stopped working.
Aslo the output of `print(sys.argv)` instead of `['mock_check/my_script.py']` now looks like this:
```
['/home/weiss-d/.cache/pypoetry/virtualenvs/mocker-m3ZuwkOU-py3.9/bin/pytest']
```
But when I run just `pytest test_my_script.py` it again passes.

This behaviour looks very strange to me, considering that every test per function should be executed isolated by Pytest, not to say about different test files.
Is it a bug, or I am missing something?

**Comment by kvas-it** [comment]
I see two things here:

1. `test_import_my_script.py` doesn't mock anything.
2. More importantly, the prints in `mock_check.my_script` are executed when it is imported, which is before any tests are executed or any fixtures are created. Even if there was mocking, it would not be in effect yet.

I think the behavior that you observe is exactly how it should be. The content of `sys.argv` is also correct. It's the `sys.argv` of the `pytest` process (which imports `mock_check.my_script` where that print is located via `test_import_my_script.py`), so it contains the path to `pytest`.

To summarize: mocking only applies while the mocking fixture is in effect, which is inside of `test_mock_check`. For the rest of the time `requests.get` is the original one and it can't get "fake.url".

Does this make sense?

**Comment by weiss-d** [comment]
Well, thanks for pointing out the import issue. I tried to simplify the example and renamed my_scrip.py. Originally it was __main__.py and contained some code not wrapped into `if __name__ == "__main__"`. While I've been blaming the mocking, the problem was that I looked right trough this unwrapped code. 🤦‍♂️ Sorry for the bustle.

**Comment by kvas-it** [comment]
No worries, it happens. BTW, if you have further problems with mocking in the scripts you might want to check out this issue #44 -- the discussion there and the linked stackoverflow thread clarify some issues related to mocking and scripts.

**Comment by weiss-d** [comment]
> No worries, it happens. BTW, if you have further problems with mocking in the scripts you might want to check out this issue #44 -- the discussion there and the linked stackoverflow thread clarify some issues related to mocking and scripts.

Funny enough, I had no problem with this one cause I got there straight from readme)
Now everything is working with mocks and imports. Big thanks for this tool!
